boomerang/src/state_machine/tests.rs

```diff
@@ -0,0 +1,73 @@
+use crate::state_machine::{
+    BoomerangSetup, DEFAULT_PRINCIPAL_ID, ONE_MONTH_SECONDS, USER_PRINCIPAL_ID,
```

Comment:
Boomerang setup feels the same as my code in waterneuron, wouldn't it make sense to reuse it?

Reply:
The struct is similar indeed. But the implementation differs so the methods are specific to boomerang. Plus I like to think of boomerang as an independent module that uses WaterNeuron that's why I did not reuse it.

Reply:
Now we have 1k lines of the exact same code. I'm pretty sure we can do better here.